pkg/kmodule/kmodule_linux.go

```diff
@@ -103,6 +113,55 @@ func Delete(name string, flags uintptr) error {
 	return unix.DeleteModule(name, int(flags))
 }
 
+// parallelProbeDep loads a module and it's dependencies
+// Returns once module is loaded.
+func parallelProbeDep(deps depMap, modPath string, params string, opts ProbeOpts) error {
+	dep, present := deps[modPath]
+	if !present {
+		return fmt.Errorf("Module not in depmap %s", modPath)
+	}
+
+	// If it's already loaded, don't try to load again
+	if dep.state == builtin || dep.state == loaded {
+		return dep.err
+	}
+
+	dep.loadOnce.Do(func() {
+		dep.state = loading
+		var eg errgroup.Group
+
+		for _, dep := range dep.deps {
+			eg.Go(func() error { return parallelProbeDep(deps, dep, "", opts) })
```

Comment:
What happens if a driver hangs forever? Is there some solution? I realize existing code probably doesn't cover that case either.